docs/source/markdown/podman-attach.1.md

```diff
@@ -9,48 +9,52 @@ podman\-attach - Attach to a running container
 **podman container attach** [*options*] *container*
 
 ## DESCRIPTION
-The attach command allows you to attach to a running container using the container's ID
-or name, either to view its ongoing output or to control it interactively.
+**podman attach** allows you to attach to a running *container* using the *container's name* or *ID*, either to view its ongoing output or to control it interactively.
 
-You can detach from the container (and leave it running) using a configurable key sequence. The default
+You can detach from the *container* (and leave it running) using a configurable key sequence. The default
 sequence is `ctrl-p,ctrl-q`.
 Configure the keys sequence using the **--detach-keys** option, or specifying
-it in the **containers.conf** file: see **containers.conf(5)** for more information.
+it in the `containers.conf` file: see [**containers.conf(5)**](https://github.com/containers/common/blob/master/docs/containers.conf.5.md) for more information.
```

Comment:
This link is showing up in the actual man page.

Reply:
Maybe we can filter them out with a regex before. In this case we could also leave the `**` in the `[]`. But sadly I am not familiar with regex.

Reply:
We would need to do some sed magic.  @edsantiago @nalind Any ideas?

Reply:
there's already some sed magic in the Makefile, for dealing with man pages. See `MANPAGES`, line 430. I'm not sure what needs to be done here but I'll take a look in the morning.

Reply:
It seems like links to other man pages or config files can only work in either markdown or in the man page. It would be really sad to dispense the idea of having commands linked to the correct .md files. But at the current stage until a solution is found for it, I will revert it.

Reply:
If I understand the problem correctly, perhaps this could solve it:
```diff
diff --git a/Makefile b/Makefile
index 15d6d9fb6..70ba76e49 100644
--- a/Makefile
+++ b/Makefile
@@ -428,8 +428,10 @@ pkg/api/swagger.yaml: .gopathok
        make -C pkg/api

 $(MANPAGES): %: %.md .install.md2man docdir
-       @sed -e 's/\((podman[^)]*\.md)\)//g' -e 's/\[\(podman[^]]*\)\]/\1/g' \
-               -e 's;<\(/\)\?\(a\|a\s\+[^>]*\|sup\)>;;g' $<  | \
+       @sed -e 's/\((podman[^)]*\.md)\)//g' \
+            -e 's/\[\(podman[^]]*\)\]/\1/g' \
+            -e 's/\[\(\*[^]]*\)\](http[^)]\+)/\1/g' \
+            -e 's;<\(/\)\?\(a\|a\s\+[^>]*\|sup\)>;;g' $<  | \
        $(GOMD2MAN) -in /dev/stdin -out $(subst source/markdown,build/man,$@)

 .PHONY: docdir
```

Reply:
I meant, that the whole link should be filtered out in the manpage. In the way it is now it's just put in a less-than sign and greater-than sign. So everything that is between the less-than sign and greater-than sign should go.

The `[]` aren't a problem anymore. 

It would also be helpful to replace the `\` at the end of the line with two spaces. In this way the desired line break can be achieved.

Reply:
>everything that is between the less-than sign and greater-than sign should go.

My patch converts `[**markup**](link)` to just `**markup**`, which isn't exactly the same thing, but has the same end effect: the `<link>` is gone from the man page:
```diff
diff -ur docs/build/man.ante/podman-attach.1 docs/build/man/podman-attach.1
--- docs/build/man.ante/podman-attach.1 2021-06-03 10:46:32.118576644 -0600
+++ docs/build/man/podman-attach.1      2021-06-03 13:50:10.629289923 -0600
@@ -19,8 +19,7 @@
 You can detach from the \fIcontainer\fP (and leave it running) using a configurable key sequence. The default
 sequence is \fB\fCctrl\-p,ctrl\-q\fR\&.
 Configure the keys sequence using the \fB\-\-detach\-keys\fP option, or specifying
-it in the \fB\fCcontainers.conf\fR file: see \fBcontainers.conf(5)\fP
-\[la]https://github.com/containers/common/blob/master/docs/containers.conf.5.md\[ra] for more information.
+it in the \fB\fCcontainers.conf\fR file: see \fBcontainers.conf(5)\fP for more information.

 .SH OPTIONS
 .SS \fB\-\-detach\-keys\fP=\fBsequence\fP
```
Is that not good enough?

Reply:
It works for http and https but no for relative links like `(podman.1.md)`. The link is gone, but the `[]` are still there, around the command.

Reply:
Quick overview for the visual errors for links in the manpages:

- `**[podman](podman.1.md)**` results in **podman**
 *for commands alone everything is working*


- `**[podman-generate-systemd --new](podman-generate-systemd.1.md#--new)**` results in 
**podman-generate-systemd --new**(podman-generate-systemd.1.md#--new)
*first possibility for command with an option*

- `[**podman-generate-systemd --new**](podman-generate-systemd.1.md#--new)` results in
 **podman-generate-systemd --new** <podman-generate-systemd.1.md#--new>
*second possibility for command with an option*
It seems like links that include an option don't work. I believe it is do to the space between the command and option.

- `**[containers.conf(5)](https://github.com/containers/common/blob/master/docs/containers.conf.5.md)**` results in
containers.conf(5) \<https://github.com/containers/common/blob/master/docs/containers.conf.5.md>
*first possibility for a link with http/https*

- `[**containers.conf(5)**](https://github.com/containers/common/blob/master/docs/containers.conf.5.md)` results in
**containers.conf(5)**
*second possibility for a link with http/https*
Since the `**` should be on the outside of the whole structure, it should be switched at the http/https links. 

Thank for the help with the regex. :)

Reply:
Can someone confirm, that the last `-e` line added to the makefile works for deleting the last two whitespaces in the `.md` file and creating a line break: 
```
$(MANPAGES): %: %.md .install.md2man docdir
	@sed -e 's/\((podman[^)]*\.md)\)//g' \
         -e 's/\[\(podman[^]]*\)\]/\1/g' \
         -e 's/\[\(\*[^]]*\)\](http[^)]\+)/\1/g' \
         -e 's;<\(/\)\?\(a\|a\s\+[^>]*\|sup\)>;;g' \
         -e 's/[[:blank:]]+/\n/g' $<  | \
	$(GOMD2MAN) -in /dev/stdin -out $(subst source/markdown,build/man,$@)
```

Reply:
Just dropping in, but would `[[:blank:]]+` not match any sequence of two or more blank characters anywhere on any line?

Reply:
Yep i think so, but my first thought was also wrong. I wanted to replace the `\` with two spaces. And i believe this works:

```
$(MANPAGES): %: %.md .install.md2man docdir
	@sed -e 's/\((podman[^)]*\.md)\)//g' \
         -e 's/\[\(podman[^]]*\)\]/\1/g' \
         -e 's/\[\(\*[^]]*\)\](http[^)]\+)/\1/g' \
         -e 's;<\(/\)\?\(a\|a\s\+[^>]*\|sup\)>;;g' \
         -e 's/\\$//  /g' $<  | \
	$(GOMD2MAN) -in /dev/stdin -out $(subst source/markdown,build/man,$@)
```

Reply:
I edited the regex a little more. It should work as intended now.